torrent flags refactor

arvidn opened the pull request:
this amends https://github.com/arvidn/libtorrent/pull/2119 with making the flags strongly typed

arvidn commented:
@AllSeeingEyeTolledEweSew would you mind taking a look at this? does it look ok?

aldenml commented on include/libtorrent/torrent_flags.hpp:
```diff
@@ -0,0 +1,267 @@
+	// indicates that this torrent should never be unloaded from RAM, even
+	// if unloading torrents are allowed in general. Setting this makes
+	// the torrent exempt from loading/unloading management.
+	constexpr torrent_flags_t TORRENT_DEPRECATED_MEMBER pinned{0x1000};;
```
there are a few double semicolon in this patch like `};;`

arvidn replied:
thanks. I fixed this

AllSeeingEyeTolledEweSew commented:
Looks good overall. I actually started writing this myself: see https://github.com/AllSeeingEyeTolledEweSew/libtorrent/tree/asetes-torrent-flags-enum-class . I branched this from my #2119 branch. I thought it might go in a separate PR, but couldn't figure out how to create a new PR based off of a pending PR (this instinct comes from big-software-house source control processes, I guess it doesn't fit github as well), so I was waiting for #2119 to land.

I had a few differences: I ended up adding `operator<<(std::ostream&, x)` to make it easier to log the flags.

Would you mind also adding a python binding for the torrent_flags class? I have one in my branch.

I kept `add_torrent_params::flags_t` separate from `torrent_flags`, but kept the numeric spaces the same. I like your approach more though; not all flags can be set in all cases (`seed_mode`) so there's no surprise that some flags only apply to `add_torrent`. May as well have a single class.

arvidn commented:
@AllSeeingEyeTolledEweSew I added the ``operator<<`` too, but I made that a separate PR (which I have rebased on top of).

This patch includes python bindings for the flags. Both the new and the deprecated ones.
I actually deprecated all of the flags in ``add_torrent_params`` and moved them into the ``torrent_flags`` namespace. You may have noticed that I also abandoned the ``enum class`` approach in favor of a proper class. again, I did that in a separate PR which this has been rebased on top of.

arvidn commented:
The reason I folder your change into a single PR is that I really think exposing the flags wider really makes the problem of just using enums more prominent, so I wanted to avoid having that at all.
